src/alerts_aggregator.py

```diff
@@ -0,0 +1,242 @@
+                time_list = self.quarter_hour_intervals_list(start_time, end_time)
+                logging.debug(f'This is the 15-minute interval time list: {time_list}')
+                adjusted_time_counts = filtered_df['adjusted_time'].value_counts().reindex(time_list, fill_value=0)
+                return adjusted_time_counts
+            
+            elif is_settlement_exist:
+                raise NoAlarmsException(f"There were no alarms in the settlement: {settlement}")   
+            else:
+                raise InvalidSettlement("You selected a Settlement that does not exist.")
+        
+        except NoAlarmsException as ex:
+            raise NoAlarmsException(ex)   
+        except InvalidSettlement as ex:
+            raise InvalidSettlement(ex)
+        except WrongSettlementException as ex:
+            raise WrongSettlementException(ex)
+        except requests.exceptions.RequestException as ex:
+            raise requests.exceptions.RequestException(ex)
+    
+        
+
+    # This function use create_quarter_hour_column func to return the best time to shower in
+    def best_time_to_shower(self, settlement : str, start_time : time, end_time : time) -> str:
+        try:    
+            # the function create_adjusted_time_column returns pd series of time object and its count value 
+            adjusted_time_counts = self.create_adjusted_time_column(settlement, start_time, end_time)
+            logging.debug(f'This is the alert distribution by 15-minute interval in {settlement}: \n{adjusted_time_counts}')
+            
+            # Detect the quarter of an hour that appeared the less-it will be the best time to shower
+            best_time = adjusted_time_counts.idxmin()
+            return f"The best time to take a shower is at: {best_time}"
+            
+        except NoAlarmsException as ex:
+            raise NoAlarmsException(ex) 
+        except InvalidSettlement as ex:
+            raise InvalidSettlement(ex)
+        except requests.exceptions.RequestException as ex:
+            raise requests.exceptions.RequestException(ex)
+   
+    
+    # This function use create_quarter_hour_column func to return the worst time to shower in
+    def worst_time_to_shower(self, settlement: str, start_time: time, end_time: time) -> str:
+        try:    
+            # the function create_adjusted_time_column returns pd series of time object and its count value 
+            adjusted_time_counts = self.create_adjusted_time_column(settlement, start_time, end_time)
+            logging.debug(f'This is the alert distribution by 15-minute interval in {settlement}: \n{adjusted_time_counts}')
+            
+            # Detect the quarter of an hour that appeared the most- it will be the worst time to shower
+            worst_time = adjusted_time_counts.idxmax()
+            return f"The worst time to take a shower is at: {worst_time}"
+            
+        except NoAlarmsException as ex:
```

Comment:
same. that module should not be aware of any user interfacing stuff.
fix the function type annotations too , function should return `time` and not `str`